### PR TITLE
Fixes #32779 - use directly smart_proxy_dynflow

### PR DESCRIPTION
**Pull request by ezr-ondrej**
The ForemanTasksCore has been discontinued and has been moved to
smart_proxy_dynflow. We should use those classes directly and not rely
on the compatibility layer tasks_core gem.

### CODE REVIEW

**Comment by adamruzicka** [comment]
smart_proxy_dynflow-0.5.0 is out

[test smart_proxy_ansible]

**Comment by adamruzicka** [comment]
smart_proxy_dynflow-0.5.1 is out

[test smart_proxy_ansible]

**Comment by adamruzicka** [comment]
Test failures are most likely related

**Comment by ezr-ondrej** [comment]
I'd love sanitized requires, but I guess this is what we get

File: lib/smart_proxy_ansible/actions.rb [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # frozen_string_literal: true
 
-require 'foreman_tasks_core/shareable_action'
+require 'smart_proxy_dynflow/action/shareable'
```
Comment (adamruzicka) [inline lib/smart_proxy_ansible/actions.rb:3]: Why do we need this here?

**Review by adamruzicka — COMMENTED** [review]
Could you also replace all `SmartProxyDynflowCore` usages?

**Comment by adamruzicka** [comment]
Needs a rebase

**Comment by adamruzicka** [comment]
Also I'm getting this when starting smart proxy

```
2021-06-15T16:48:29  [W] Error details for Couldn't enable 'ansible': <NameError>: uninitialized constant Proxy::Dynflow::Runner::CommandRunner
smart-proxy-ansible/lib/smart_proxy_ansible/remote_execution_core/ansible_runner.rb:6:in `<module:RemoteExecutionCore>'
smart-proxy-ansible/lib/smart_proxy_ansible/remote_execution_core/ansible_runner.rb:4:in `<module:Ansible>'
smart-proxy-ansible/lib/smart_proxy_ansible/remote_execution_core/ansible_runner.rb:3:in `<top (required)>'
smart-proxy-ansible/lib/smart_proxy_ansible/plugin.rb:17:in `require'
smart-proxy-ansible/lib/smart_proxy_ansible/plugin.rb:17:in `block in <class:Plugin>'
```

File: lib/smart_proxy_ansible/actions.rb (hunk 1 above)
Comment (ezr-ondrej) [reply lib/smart_proxy_ansible/actions.rb:3]: no idea why it was here, requires are very weird all over the place... I'll try to make them bit more sensible

**Comment by ezr-ondrej** [comment]
I'm very unhappy about our requires xD but this loads :thinking:

**Comment by adamruzicka** [comment]
Thank you @ezr-ondrej !

**Pull request merged**